doc/built-in-extensions.rst

```diff
@@ -6,6 +6,27 @@ Built-In Extensions
 Hawkmoth is :ref:`extensible <extending>`, and ships with some built-in
 extensions.
 
+.. _hawkmoth.ext.javadoc:
+
+hawkmoth.ext.javadoc
+--------------------
+
+This extension converts Javadoc_ comments to reStructuredText.
+
+.. _Javadoc: https://www.oracle.com/technetwork/java/javase/documentation/javadoc-137458.html
+
+Installation and configuration in ``conf.py``:
+
+.. code-block:: python
+
+   extensions.append('hawkmoth.ext.javadoc')
+
+.. py:data:: hawkmoth_javadoc_transform
+   :type: list
+
+   List of transforms to convert, defaults to ``['javadoc']``. If the
+   ``transform`` option is not in in this list, do nothing.
```

Comment:
Hmm, this reads a bit funny and looks more complicated than it should. What would be the use case for the list?

`transform` is to be a string that needs to match the right trigger for this extension. Even if we make that trigger configurable (maybe a good idea as long as we support per-directive transformations), it should be a string and the text should read something like 'the name to use for this transformation'.

Same for the napoleon one.

Reply:
Yeah, so maybe this isn't fully thought out yet. The idea is, you've got documentation with `:transform: something-or-other` all over the place, and want to switch to using the built-in extension. But if it only accepts `javadoc` and nothing else, you're going to have to replace everything in flag day style.

Granted, looks like the upstream autodoc doesn't have anything like this. Even napoleon opportunistically converts if it can, and there's no further configuration, just adding `sphinx.ext.napoleon` to `extensions`.

Reply:
Registering the exact same transformation under multiple keys doesn't strike me as something we want to / should support in new APIs and I'd be surprised if anyone found / would find a reasonable use case for such a thing.

Still, there's already an extension providing backwards compatibility, why compromise the new stuff?

I'd say:
1. Keep supporting per directive, explicit transformations despite what napoleon does for instance. I think it's valuable and it's simple enough.
   The only thing here that makes me question this is that 3rd party plugins are not constrained to respect `:transform:` and I can't see a way of enforcing it without gating the emission of the event, which would be even worse I think.
3. Being a new API, keep it free from old stuff and as simple / elegant as possible. I'd argue the trigger for our built in extensions should _not_ be configurable at all until we find a strong reason for it and definitely not a list.

Reply:
Converted the transform names to strings in #131, and did the matching with simple `!=`. Clarified documentation as well.

I'm still keeping the simple configuration options though, I hope that's fine.